lib/Doctrine/ORM/UnitOfWork.php

```diff
@@ -2401,6 +2401,13 @@ public function clear($entityName = null)
                     $this->doDetach($entity, $visited, false);
                 }
             }
+
+            foreach ($this->entityInsertions as $hash => $entity) {
+                if (get_class($entity) != $entityName) {
+                    continue;
```

Comment:
In this scenario, using `continue` is more confusing, IMO. `unset()` can live inside the `if()` if you don't negate the comparison